lib/smf/smf.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2021 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "smf.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(smf);
+
+/*
+ * Private structure (to this file) used to track state machine context.
+ * The structure is not used directly, but instead to cast the "internal"
+ * member of the smf_ctx structure.
+ */
+struct internal_ctx {
+#ifdef CONFIG_SMF_ANCESTOR_SUPPORT
+	bool new_state : 1;
+#endif
```

Comment:
If you allow the new_state flag unconditionally, I believe you can remove the #ifdef around the ``smf_execute_ancestor*`` routines.  You can then also convert all the calls to the ``smf_execute_ancestor*`` routines using IS_ENABLED(CONFIG_SMF_ANCESTOR_SUPPORT).

You'll have to add the __unused attribute to these routines to avoid a warning when CONFIG_SMF_ANCESTOR_SUPPORT=n

Reply:
Done